### ISSUE TITLE
Recursive macro expansion is uninterruptible

### ISSUE THREAD

**Issue by martinwguy** [martinwguy/xvi#45]
```
:map a b
:map b a

```
sends xvi into an infinite loop when you press a or b

classic vi says it only remember b mapped to a but the output of :map is followed by "ERROR", "a" appends, and so does "b".
vim remembers both, but when you press either says "E223: recursive mapping" and stops.
nvi remembers both and goes into an infinite loop when you press a or b.

The "right" behaviour is therefore arguable,,,

To make it interruptible, the place to insert a test seems to be here:
```
--- a/src/map.c
+++ b/src/map.c
@@ -153,6 +153,12 @@ map_getc()
      * awaiting translation in the canonical queue.
      */
     while (flexempty(npos.mp_dest)) {
+       /* Allow them to interrupt loops due to recursive remapping */
+       if (kbdintr) {
+           flexclear(&canon_queue);
+           return(CTRL('C'));
+       }
        if (!flexempty(npos.mp_src)) {
            mapthrough(&npos,
                        (State == NORMAL) ? cmd_map :
```
This stops and prints "Interrupted" if you were in normal and cmdline mode and in insert/replace mode with ```:map! x x[Enter]ax```
but if you were in insert or replace mode (using ```:map! x bx[Enter]ax```) it stops but doesn't show "Interrupted".

**Comment by martinwguy** [comment]
Seems to be fixed, i-e- it is now interruptible

**Comment by martinwguy** [comment]
Nope. Still present.

**Comment by martinwguy** [comment]
Fixed by 1ec9b6a2
